src/rules/noTrailingWhitespaceRule.ts

```diff
@@ -72,53 +73,48 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 function walk(ctx: Lint.WalkContext<IgnoreOption>) {
-    let lastSeenWasWhitespace = false;
-    let lastSeenWhitespacePosition = 0;
-    utils.forEachTokenWithTrivia(ctx.sourceFile, (fullText, kind, range) => {
-        if (kind === ts.SyntaxKind.NewLineTrivia || kind === ts.SyntaxKind.EndOfFileToken) {
-            if (lastSeenWasWhitespace) {
-                reportFailure(ctx, lastSeenWhitespacePosition, range.pos);
+    const possibleFailures: ts.TextRange[] = [];
+    const sourceFile = ctx.sourceFile;
+    const text = sourceFile.text;
+    for (const line of getLineRanges(sourceFile)) {
+        const match = text.substr(line.pos, line.contentLength).match(/\s+$/);
+        if (match !== null) {
+            possibleFailures.push({
+                end: line.pos + line.contentLength,
```

Comment:
This is why I don't like `object-literal-sort-keys`: It doesn't make sense to have the keys in alphabetical order here.